### PR TITLE
DOC: add notes about the recently merged store backend feature

### PR DESCRIPTION
**Pull request by aabadie**
I realized that the new API and internal changes of Memory were not added to the changes.rst file.

This PR provides the update.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -27,6 +27,19 @@ Olivier Grisel
...
+Alexandre Abadie
+
+    Introduce the concept of 'store' and refactor the ``Memory`` internal
+    storage implementation to make it accept extra store backends for caching
+    results. ``backend`` and ``backend_options`` are the new options added to
+    ``Memory`` to specify and configure a store backend.
+
+    Add the ``register_store_backend`` function to extend the store backend
+    used by default with Memory. This default store backend is named 'local'
+    and corresponds to the local filesystem.
```
Comment (ogrisel) [inline CHANGES.rst]: Maybe add a note to say that the store backend API itself is experimental and subject to change without deprecation.

Comment (aabadie) [reply CHANGES.rst]: ok

**Comment by aabadie** [comment]
sorry, I accidentally removed the branch on my remote...

File: CHANGES.rst (hunk 1 above)
Comment (aabadie) [reply CHANGES.rst]: note added

**Comment by GaelVaroquaux** [comment]
LGTM. Thank you! Merging!

**Pull request merged**